### PR TITLE
Conda updates

### PR DESCRIPTION
**Pull request by richafrank**
Updated versions of conda tooling

Install newer miniconda - This fixes the failing appveyor build on master.

Removed python 3.4 builds

### CODE REVIEW

File: conda/parameterized/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
+{% set name = "parameterized" %}
+{% set version = "0.6.1" %}
+{% set file_ext = "tar.gz" %}
+{% set hash_type = "sha256" %}
+{% set hash_value = "caf58e717097735de0d7e15386a46ffa5ce25bb6a13a43716a8854a8d34841e2" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
```
Comment (freddiev4) [inline conda/parameterized/meta.yaml]: Where does `lower` come from? Does that signify that when the packages are built then conda will read the `name` as lowercase?

File: ci/make_conda_packages.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def main(env, do_upload):
                "--python", env['CONDA_PY'],
                "--numpy", env['CONDA_NPY'],
                "--skip-existing",
+               "--old-build-string",
```
Comment (freddiev4) [inline ci/make_conda_packages.py]: 👍 for adding this arg

File: appveyor.yml [hunk 3]
```diff
@@ -78,7 +68,8 @@ install:
   - sed -i "s/scipy==.*/scipy==%SCIPY_VERSION%/" etc/requirements.txt
 
   - conda info -a
-  - conda install conda=4.1.11 conda-build=1.21.11 anaconda-client=1.5.1 --yes -q
+  - conda install conda=4.3.30 conda-build=3.0.28 anaconda-client=1.6.3 --yes -q
```
Comment (freddiev4) [inline appveyor.yml]: Are there any many main differences/reasons for using these newer versions, aside from fixing the build?

File: .travis.yml [hunk 4]
```diff
@@ -3,7 +3,6 @@ sudo: false
 fast_finish: true
 python:
   - 2.7
-  - 3.4
```
Comment (freddiev4) [inline .travis.yml]: Should we still run tests for Py34?

File: conda/0_sortedcontainers/build.sh [hunk 5]
```diff
@@ -1,9 +0,0 @@
-#!/bin/bash
-
-$PYTHON setup.py install
-
-# Add more build steps here, if they are necessary.
-
-# See
-# http://docs.continuum.io/conda/build.html
-# for a list of environment variables that are set during the build process.
```
Comment (freddiev4) [inline conda/0_sortedcontainers/build.sh]: TIL you can move this into the `.yaml` files. That's cool!

File: conda/parameterized/meta.yaml (hunk 1 above)
Comment (richafrank) [reply conda/parameterized/meta.yaml]: > Where does `lower` come from? 

It's built in to the [jinja2](http://jinja.pocoo.org/docs/2.9/templates/#lower) interpreter used to handle the [templating](https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#templating-with-jinja) here.

> Does that signify that when the packages are built then conda will read the `name` as lowercase?

Yea, the `name` variable (defined above statically as `"parameterized"`) will be lowered before being used as the name of the package. Given that static name, lowering is a noop here, but this is all autogenerated by `conda skeleton`.

File: conda/0_sortedcontainers/build.sh (hunk 5 above)
Comment (richafrank) [reply conda/0_sortedcontainers/build.sh]: Yep, that's from the newer version of conda-build. I used `conda skeleton` to re-generate these recipes.

File: appveyor.yml (hunk 3 above)
Comment (richafrank) [reply appveyor.yml]: There are definitely new features and bug fixes. Many are optional, like the new build strings and changes in how one pins dependencies (like the numpy C-API), which I haven't tried out. You noticed the consolidated recipes - I regenerated a bunch of ours.

The reason I updated in this PR was to fix our toolchain, in that the old miniconda could no longer work with the package index or update to newer conda versions.

File: .travis.yml (hunk 4 above)
Comment (richafrank) [reply .travis.yml]: I was seeing build failures from missing dependencies in py34, but we're no longer running anything on py34, so it's ok to remove.

**Comment by freddiev4** [comment]
After reading through the responses, LGTM 👍

**Pull request merged**